fili-core/src/main/java/com/yahoo/bard/webservice/data/config/provider/PhysicalTableConfiguration.java

```diff
@@ -0,0 +1,31 @@
+// Copyright 2016 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.data.config.provider;
+
+import com.yahoo.bard.webservice.data.config.dimension.DimensionConfig;
+import com.yahoo.bard.webservice.data.config.names.FieldName;
+import com.yahoo.bard.webservice.data.config.table.PhysicalTableDefinition;
+
+import java.util.Set;
+
+/**
+ * A PhysicalTableConfiguration can create its PhysicalTableDefinition, given a dimension configuration.
+ */
+public interface PhysicalTableConfiguration {
+
```

Comment:
It looks like this interface is a a PhysicalDefinitionFactory.  But you're calling it a PhysicalTableConfiguration.
As best as I can infer, this is going to be backed by a 'smart bean', where the underlying bean is initialized from a configuration model (e.g. YAML file, xml, database query, etc), and this INTERFACE discusses using that model AS a factory.

Maybe this is a conceptual confusion, or maybe this is a naming issue.

If you have an object modeling a thing that provides name, grain, dim names, metric names, maybe that bean should be an explicit thing. (e.g. a PhysicalTableConfiguration) and there should be another thing that implements what you've described above pulling the parts out of the model object and building a definition out of it.